### PR TITLE
Update Kill exceptions

### LINKED ISSUE #6966: Process.Kill() behavior changed around core3.1 release but documentation fails to reflect that
Here's the cause of change https://github.com/dotnet/runtime/issues/28672 It was implemented around core 3.1 release. Here's the documentation for 5.0: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=net-5.0 - it still mentions the old behavior ("The process has already exited." option).

This change needs to be documented in all versions where it is in effect. It would also be nice if a "breaking change" note was included because this change makes some error-handling code in user codebases redundant.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #6966.

### CODE REVIEW

File: xml/System.Diagnostics/Process.xml [hunk 1]
```diff
@@ -2328,16 +2328,16 @@ You cannot cause processes on remote computers to exit. You can only view inform
   
  The process is terminating.</exception>
         <exception cref="T:System.NotSupportedException">You are attempting to call <see cref="M:System.Diagnostics.Process.Kill" /> for a process that is running on a remote computer. The method is available only for processes running on the local computer.</exception>
-        <exception cref="T:System.InvalidOperationException">The process has already exited.  
+        <exception cref="T:System.InvalidOperationException">.NET Framework and .NET Core 3.0 and earlier versions only: The process has already exited.  
   
 -or-  
 
 There is no process associated with this <see cref="T:System.Diagnostics.Process" /> object.
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System.Diagnostics/Process.xml:2428]: It seems this code has been refactored a bit in the newer verisons, but looks like it still checks the state of the process (exited) and throws accordingly: [runtime/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L945](https://github.com/dotnet/runtime/blob/983ff47fa33e1c7d452f594db7584667622c2604/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L945 )

File: xml/System.Diagnostics/Process.xml [hunk 2]
```diff
@@ -2421,16 +2421,16 @@ You cannot cause processes on remote computers to exit. You can only view inform
...
 
 -or-
 
 There is no process associated with this <see cref="T:System.Diagnostics.Process" /> object.
 
 -or-
 
-The calling process is a member of the associated process' descendant tree.</exception>
-        <exception cref="T:System.AggregateException">Not all processes in the associated process' descendant tree could be terminated.</exception>
+The calling process is a member of the associated process's descendant tree.</exception>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/Process.xml:2432]: TIL something new. I always thought that only `s'` was valid. Seems that both variants of apostrophe for a word ending in "s" are equally valid: 

> For names that end in an s or z sound, though, you can either add -'s or just an apostrophe. Going with -'s is the more common choice:
> the car that belongs to Jones → Jones's car or Jones' car

Source: https://www.merriam-webster.com/words-at-play/what-happens-to-names-when-we-make-them-plural-or-possessive

Comment (gewarren) [reply xml/System.Diagnostics/Process.xml:2432]: Yes, and MS style guide says to use 's too: https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28753

File: xml/System.Diagnostics/Process.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gewarren) [reply xml/System.Diagnostics/Process.xml:2428]: @carlossanlop This is specifically about the Kill method - see https://github.com/dotnet/runtime/blob/983ff47fa33e1c7d452f594db7584667622c2604/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L104.

[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System.Diagnostics/Process.xml:2428]: Yes. Sorry, I linked a line to `EnsureState`. I meant to link to the line of `GetHasExited`. Both were refactored in the corefx PR, and the code remains very similar in runtime: `Kill` will not throw if the process has exited.

**Comment by gewarren** [comment]
@carlossanlop Can you approve if you like it?

**Review by carlossanlop — APPROVED** [review]
Approving only because I liked it.

**Pull request merged**